src/ebay-pagination/helpers.ts

```diff
@@ -1,18 +1,21 @@
-import { MAX_PAGES, MIN_PAGES } from './const'
+import { MAX_PAGES, MIN_PAGES, SHOW_LAST, OVERFLOW, TRAILING_SPACE_WITH_DOT, LEADING_SPACE_WITH_DOT } from './const'
 import { ItemState, PaginationVariant } from './types'
 
 export function pageNumbersAround(
     totalPages: number,
     selectedPage: number,
     maxVisiblePages: number = totalPages,
-    withDots = false
+    variant: PaginationVariant = null
 ): ItemState[] {
+    const withDots = variant === SHOW_LAST || (variant === OVERFLOW && totalPages > MAX_PAGES)
+    const hasLeadingDots = variant === OVERFLOW && totalPages > MAX_PAGES
     const visibleItems = Math.min(maxVisiblePages, totalPages)
     const startIndexWithoutDots = Math.max(0, selectedPage - Math.ceil((visibleItems - 1) / 2))
     const startIndexWithDots = visibleItems < 4 ? selectedPage :
         Math.max(0, selectedPage - Math.floor((visibleItems - 1) / 2))
     const endIndex = (withDots ? startIndexWithDots : startIndexWithoutDots) + visibleItems
     const closeToEnd = endIndex >= totalPages
+    const closeToFront = selectedPage <= 4
```

Comment:
Why `4`? If it is related to line 14, can we add them in a constant?

Reply:
sure will do

Reply:
updated